openedx/frontend-app-course-authoring
# feat: xblock status component and copy & paste unit option

**Description:**

Adds status text under each item in the outline. Screenshot from legacy UI for reference:
![image](https://github.com/open-craft/frontend-app-course-authoring/assets/10894099/8b31c505-7762-431f-a5a7-f0dc1b439875)

The reference logic for displaying this text can be found in https://github.com/openedx/edx-platform/blob/master/cms/templates/js/course-outline.underscore.

**Also combines** https://github.com/open-craft/frontend-app-course-authoring/pull/29

<details>
  <summary>MFE Screenshot:</summary>
 

![image](https://github.com/open-craft/frontend-app-course-authoring/assets/10894099/390b9185-18dd-497f-88ee-eee983772089)


</details> 

**Test instructions:**

* Get devstack up and running using `make {lms,cms}-up`.
* Start npm devserver in this repo using `npm start`.
* Goto http://localhost:2001/course/course-v1:edX+DemoX+Demo_Course/
* Different types of status text should be visible based on the item configuration.
* You can enable proctoring exam via advanced settings page.

**Depends on:** https://github.com/openedx/edx-platform/pull/34058
